### ISSUE TITLE
Vagrant provision order 

### ISSUE THREAD

**Issue by Temikus** [mitchellh/vagrant-google#87]
We currently have the following Provision action:

```
      # This action is called when `vagrant provision` is called.
      def self.action_provision
        Vagrant::Action::Builder.new.tap do |b|
          b.use ConfigValidate
          b.use Call, IsCreated do |env, b2|
            if !env[:result]
              b2.use MessageNotCreated
              next
            end

            b2.use Provision
            b2.use SyncedFolders
          end
        end
      end
```

Logic seems a bit weird, since it syncs the folders only _after_ the provision.
Need to check:
- The step order to make sure we can actually provision with a changed script.
- Whether SyncedFolders action is actually needed, since vagrant-aws for example, got rid of theirs:
  https://github.com/mitchellh/vagrant-aws/commit/df57c6a687c33e85ac413eae7e240f805b241c1c

**Comment by Temikus** [comment]
"The rsync and rsync-auto commands can be used to force a resync and to automatically resync when changes occur in the filesystem. Without running these commands, **Vagrant only syncs the folders on vagrant up or vagrant reload.**"
This means that to comply to the provider standard, we need to remove the SyncedFolders action from provision altogether.
However, this is not fun for people testing, since this will mean that one needs to run `rsync-auto` before provisioning or `vagrant rsync && vagrant provision` every time. There are issues about it:
https://github.com/mitchellh/vagrant/issues/5850
berkshelf/vagrant-berkshelf#281
berkshelf/vagrant-berkshelf#284
mitchellh/vagrant-aws#385

/CC @erjohnso 
Eric, what do you think? I see to options (leaning towards the second one):
- Comply with provider standard and wait until --rsync (see above) is implemented?
- Put the SyncedFolders before Provision for the time being, risking slight non-compliance?

**Comment by Temikus** [comment]
Okay, looks like this needs to be figured out with upstream first, see last comment in: https://github.com/mitchellh/vagrant-aws/issues/385
> #347 may need to be reverted as this is is major regression in relied upon behavior, regardless of what the docs have claimed.. Current master seriously breaks the work-flow with the puppet provisioner and related plugins such as vagrant-librarian-puppet. Prior to #347, all that was requires was to run vagrant provision <foo>. Post, you have to run vagrant provision <foo> in order to trigger librarian-puppet and wait through an unneeded puppet run, then run vagrant rsync <foo> to copy the modules over to the box, and then run vagrant provision <foo> again, this time waiting through an unnecessary librarian-puppet update before the "effective" puppet run takes place.

**Comment by Temikus** [comment]
Fixed in 2.0
